pandas/tests/frame/indexing/test_xs.py

```diff
@@ -319,3 +319,11 @@ def test_xs_droplevel_false(self):
         result = df.xs("a", axis=1, drop_level=False)
         expected = DataFrame({"a": [1]})
         tm.assert_frame_equal(result, expected)
+
+    def test_xs_droplevel_false_view(self):
+        # GH#37832
+        df = DataFrame([[1, 2, 3]], columns=Index(["a", "b", "c"]))
+        result = df.xs("a", axis=1, drop_level=False)
+        df["a"] = 2
```

Comment:
this is tricky, better to do the set on df.values to be on the safe side

Reply:
Thx, done